backbone/src/main/java/org/researchstack/backbone/ui/task/ReviewStepFullScreenImageActivity.kt

```diff
@@ -0,0 +1,77 @@
+package org.researchstack.backbone.ui.task
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.ViewTreeObserver
+import androidx.lifecycle.Observer
+import kotlinx.android.synthetic.main.rsb_activity_full_screen_image.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+import org.researchstack.backbone.R
+import org.researchstack.backbone.step.Step
+import org.researchstack.backbone.ui.PinCodeActivity
+import org.researchstack.backbone.ui.step.fragments.ReviewStepFragment
+
+
+class ReviewStepFullScreenImageActivity : PinCodeActivity(), View.OnClickListener {
+    private val viewModel: ReviewStepFullScreenImageViewModel by viewModel { parametersOf(intent) }
+
+    companion object {
+        private const val EXTRA_STEP = "full_screen_step"
+        internal const val EXTRA_IMAGE_URL = "full_screen_image_url"
+        fun getCallingIntent(context: Context, step: Step, imageUrl: String): Intent {
+            val intent = Intent(context, ReviewStepFullScreenImageActivity::class.java)
+            intent.putExtra(EXTRA_STEP, step)
+            intent.putExtra(EXTRA_IMAGE_URL, imageUrl)
+            return intent
+        }
+
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.rsb_activity_full_screen_image)
+
+        initCallbacks()
+    }
+
+    private fun initCallbacks() {
+        viewModel.displayImageEvent.observe(this, Observer {
+            fullScreenImage.setImageBitmap(it)
+        })
+        fullScreenImageClose.setOnClickListener(this)
+        fullScreenImageEditStep.setOnClickListener(this)
+
+        fullScreenImageContainer.viewTreeObserver.addOnGlobalLayoutListener(object :
```

Comment:
Chris Bane (from Google) once said: 99% of the times, using viewTreeObserver is "massively overkill" ([source](https://twitter.com/chrisbanes/status/986171606077550592)). Can you try to see if there's a simpler/better way to do this? Perhaps move the code onStart? If not, this is fine, although *if the project has KTX-Extensions* (can't remember now), perhaps you can try to see if the android-ktx extension `doOnLayout()` or similar is enough? That'd onlyl be applied to your view (as opposed to the entire layout).

Reply:
We aren't using ktx in this project, so I'm gonna use `addOnLayoutChangeListener` instead.